pkg/agent/bgp/engine/gobgp.go

```diff
@@ -0,0 +1,295 @@
+	for i := range peerConfigs {
+		request := &gobgpapi.AddPeerRequest{Peer: toGoBGPPeer(&peerConfigs[i], ipFamily)}
+		if err := g.server.AddPeer(context.TODO(), request); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (g *GoBGPServer) UpdatePeers(peerConfigs []PeerConfig, ipFamily net.IPFamily) error {
+	for i := range peerConfigs {
+		request := &gobgpapi.UpdatePeerRequest{Peer: toGoBGPPeer(&peerConfigs[i], ipFamily)}
+		if _, err := g.server.UpdatePeer(context.TODO(), request); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (g *GoBGPServer) RemovePeers(peerConfigs []PeerConfig) error {
+	for _, peerConfig := range peerConfigs {
+		request := &gobgpapi.DeletePeerRequest{Address: peerConfig.Address}
+		if err := g.server.DeletePeer(context.TODO(), request); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (g *GoBGPServer) GetPeers() ([]PeerStatus, error) {
+	var peerStatuses []PeerStatus
+
+	fn := func(peer *gobgpapi.Peer) {
+		if peer == nil {
+			return
+		}
+		peerStatus := PeerStatus{}
+		if peer.Transport != nil {
+			peerStatus.Port = int32(peer.GetTransport().GetRemotePort())
+		}
+		if peer.Conf != nil {
+			peerStatus.Address = peer.GetConf().GetNeighborAddress()
+			peerStatus.ASN = int32(peer.GetConf().GetPeerAsn())
+		}
+		if peer.GetEbgpMultihop() != nil && peer.GetEbgpMultihop().GetEnabled() {
+			peerStatus.MultihopTTL = int32(peer.GetEbgpMultihop().GetMultihopTtl())
+		} else {
+			peerStatus.MultihopTTL = DefaultBGPMultihopTTL
+		}
+		if peer.GetGracefulRestart() != nil {
+			peerStatus.GracefulRestartTimeSeconds = int32(peer.GetGracefulRestart().GetRestartTime())
+		}
+		if peer.State != nil {
+			peerStatus.SessionState = toSessionState(peer.GetState().GetSessionState())
+			if peerStatus.SessionState == SessionEstablished && peer.GetTimers() != nil && peer.GetTimers().GetState() != nil {
+				peerStatus.UptimeSeconds = int(time.Since(peer.GetTimers().GetState().GetUptime().AsTime()).Seconds())
+			}
+		}
+		peerStatuses = append(peerStatuses, peerStatus)
+	}
+
+	request := &gobgpapi.ListPeerRequest{EnableAdvertised: true}
+	err := g.server.ListPeer(context.TODO(), request, fn)
+	if err != nil {
+		return peerStatuses, err
+	}
+	return peerStatuses, nil
+}
+
+func (g *GoBGPServer) AdvertiseRoutes(routes []Route, ipFamily net.IPFamily) error {
+	for i := range routes {
+		request := &gobgpapi.AddPathRequest{Path: toGoBGPPath(&routes[i], ipFamily)}
+		if _, err := g.server.AddPath(context.TODO(), request); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (g *GoBGPServer) WithdrawRoutes(routes []Route, ipFamily net.IPFamily) error {
+	for i := range routes {
+		request := &gobgpapi.DeletePathRequest{Path: toGoBGPPath(&routes[i], ipFamily)}
+		if err := g.server.DeletePath(context.TODO(), request); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (g *GoBGPServer) GetRoutes(routeType RouteType, peerAddress string, ipFamily net.IPFamily) ([]Route, error) {
```

Comment:
I don't find its caller in the other PR.

Reply:
The same as https://github.com/antrea-io/antrea/pull/6447#discussion_r1647706297